src/Build/BackEnd/Shared/CircularDependencyException.cs

```diff
@@ -31,7 +31,7 @@ internal CircularDependencyException(string message)
         /// <summary>
         /// Constructor for deserialization.
         /// </summary>
-        protected CircularDependencyException(SerializationInfo info, StreamingContext context)
+        public CircularDependencyException(SerializationInfo info, StreamingContext context)
```

Comment:
Is this change necessary? I think it's ok for this ideally-unused constructor to continue to be protected.